civihr_employee_portal/js/filters.js

```diff
@@ -25,63 +25,33 @@ Drupal.behaviors.civihr_employee_portal_filters = {
                     var status_message = 'The whole leave request has been Rejected.';
                 }
 
-                CRM.api3('Activity', 'get', {
-                    "sequential": 1,
-                    "source_record_id": clicked_object,
-                }).done(function(result) {
-
-                    for (index = 0; index < result.count; ++index) {
-
-                        CRM.api3('Activity', 'setvalue', {
-                            "sequential": 1,
-                            "id": result.values[index].id,
-                            "field": "status_id",
-                            "value": status_value
-                        }).done(function(result) {
-
-                        });
-                    }
-
-                });
-
-                CRM.api3('Activity', 'setvalue', {
-                    "sequential": 1,
-                    "id": clicked_object,
-                    "field": "status_id",
-                    "value": status_value
-                }).done(function(result) {
-
-                    if (result.is_error == 1) {
-                        swal("Failed!", result.error_message, "error");
-                    }
-                    else {
-
-                        // Update absence status on the screen
-                        $("#act-id-" + clicked_object).html(status_type);
-
-                        // Remove the row (so it will not be loaded when rebuilding the filters)
-                        $("#act-id-" + clicked_object).closest('tr').remove();
+                $.ajax({
+                    type: 'GET',
+                    url: Drupal.settings.basePath + 'manager_approval/ajax/quick_approval/' + clicked_object + '/' + status_value + '/' + type,
+                    success: function(data) {
+                        console.log('Decision processed on Leave: ' + type);
+                        if (data.status == 'error') {
+                            swal("Failed!", data.error, "error");
+                        }
+                        else {
+                            // Update absence status on the screen
+                            $("#act-id-" + clicked_object).html(status_type);
 
-                        // Rebuild filters
-                        loadFilters();
+                            // Remove the row (so it will not be loaded when rebuilding the filters)
+                            $("#act-id-" + clicked_object).closest('tr').remove();
 
-                        console.log(Drupal.settings.basePath + 'ajax/quick_email_notify/' + clicked_object + '/' + type);
+                            // Rebuild filters
+                            loadFilters();
 
-                        // Notify by email
```

Comment:
this is important to send a notification via the email you should not delete it .. I Think it will be even better to approve and send the email in a single ajax call instead of two .. so you maybe can keep it removed and inside **civihr_employee_portal_manager_approval_quick_approval** drupal function you can call this function this function **_ajax_quick_email_notify**  to send the email